lib/segment/src/index/hnsw_index/gpu/gpu_graph_builder.rs

```diff
@@ -145,26 +133,41 @@ mod tests {
         )
         .unwrap();
 
-        let ids = (0..num_vectors as PointOffsetType).collect();
-        build_hnsw_on_gpu(
+        let mut gpu_search_context = GpuInsertContext::new(
             &gpu_vector_storage,
-            &test.graph_layers_builder,
             groups_count,
-            1,
-            cpu_linked_points_count,
+            test.graph_layers_builder.m(),
+            test.graph_layers_builder.m0(),
+            test.graph_layers_builder.ef_construct(),
             exact,
-            ids,
-            |point_id| {
-                let added_vector = test
-                    .vector_holder
-                    .vectors
-                    .get(point_id as VectorOffsetType)
-                    .to_vec();
-                Ok(test.vector_holder.get_scorer(added_vector.clone()))
-            },
-            &false.into(),
+            1..GPU_MAX_VISITED_FLAGS_FACTOR,
```

Comment:
I agree, this seems wrong:

```suggestion
            1..=GPU_MAX_VISITED_FLAGS_FACTOR,
```

or 

```suggestion
            0..GPU_MAX_VISITED_FLAGS_FACTOR,
```

Reply:
Fixed